common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -238,8 +244,74 @@ def student_view(self, context):
+        could short circuit rendering after this section.
+        """
+        newrelic.agent.add_custom_parameter('seq.block_id', unicode(self.location))
+        newrelic.agent.add_custom_parameter('seq.display_name', self.display_name or '')
+        newrelic.agent.add_custom_parameter('seq.position', self.position)
+        newrelic.agent.add_custom_parameter('seq.is_time_limited', self.is_time_limited)
+
+    def _capture_full_seq_item_metrics(self, display_items):
+        """
+        Capture information about the number and types of XBlock content in
+        the sequence as a whole. We send this information to New Relic so that
+        we can do better performance analysis of courseware.
+        """
+        # Basic count of the number of Units (a.k.a. VerticalBlocks) we have in
+        # this learning sequence
+        newrelic.agent.add_custom_parameter('seq.num_units', len(display_items))
+
+        # Count of all modules (leaf nodes) in this sequence (e.g. videos,
+        # problems, etc.) The units (verticals) themselves are not counted.
+        all_item_keys = list(
+            itertools.chain.from_iterable(
+                unit.children for unit in display_items if hasattr(unit, 'children')
+            )
+        )
+        newrelic.agent.add_custom_parameter('seq.num_items', len(all_item_keys))
+
+        # Count of all modules by block_type (e.g. "video": 2, "discussion": 4)
+        block_counts = collections.Counter(usage_key.block_type for usage_key in all_item_keys)
+        for block_type, count in block_counts.items():
+            newrelic.agent.add_custom_parameter('seq.block_counts.{}'.format(block_type), count)
+
+    def _capture_current_unit_metrics(self, display_items):
+        """
+        Positions are stored with indexing starting at 1. If we get into a
+        weird state where the saved position is out of bounds (e.g. the
+        content was changed), avoid going into any details about this unit.
+        """
+        if 1 <= self.position <= len(display_items):
+            current = display_items[self.position - 1]
+            newrelic.agent.add_custom_parameter('seq.current.block_id', unicode(current.location))
+            newrelic.agent.add_custom_parameter('seq.current.display_name', current.display_name or '')
+
+            # This is in case we have a sequence that doesn't follow the normal
+            # pattern -- e.g. an HTML module as a direct child of a Sequence. It
+            # almost never happens at the moment since it's impossible to do in
+            # Studio and we don't really test that it works properly in LMS.
+            # That being said, we check to see if there are children before
+            # doing any manipulations so that we don't crash novel course
+            # structures. If it turns out that these course structures are more
+            # popular, we can do a more general solution here.
+            if hasattr(current, 'children'):
```

Comment:
Would it make more sense to filter explicitly on `verticals` rather than just on children of sequentials that have children themselves? A cursory glance in outdated coursegraph data shows the following children of sequentials that may have `children` set on them:
`split_test`
`p` (not a typo)
`conditional`
`randomize`